### DISCUSSION TITLE
Tracks available on Musicbrainz and Spotify websites are missing in beets

### DISCUSSION THREAD

**Discussion by fetkitty** [beetbox/beets#4187] [Q&A] [answered]
When trying to search for a song I do get the corresponding album as a search result, however, the exact song I want to use is missing in beets. If I click on the provided link, the songs are available on the website. Here is an example:

 
```
C:\Users\User\Music\flac\Cham! (1 items)
Finding tags for album "マクロスMACROSS 82-99 - CHAM!".
Candidates:
1. マクロスMACROSS 82–99 - CHAM! (59.2%) (missing tracks) (Digital Media, 2017, XW, Neoncity Records)
2. マクロスMACROSS 82–99 - Aogashima Island ! (43.6%) (id, album, tracks) (Digital Media, 2017, XW)
3. マクロスMACROSS 82–99 - Sakura (41.0%) (id, album, tracks, ...) (Digital Media, 2017, XW, Banana Network)
4. マクロスMACROSS 82–99 - CHAM! (39.6%) (missing tracks, id, year) (Digital Media, 2015, XW)
5. マクロスMACROSS 82–99 - CHAM! (36.3%) (missing tracks, id, label, ...) (Digital Media, 2015, XW, DaFuture, DFTR008)
```
(note the missing tracks annotation next to the first result)

If I then try to open the first result I get the following:

```
URL:
https://musicbrainz.org/release/bf96f898-83c6-48cc-90bd-4ddf493af9a9
(Similarity: 59.2%) (missing tracks) (Digital Media, 2017, XW, Neoncity Records)
Missing tracks (13/14 - 92.9%):
 ! Intro (w Kokayna)                      (# 1) (0:36)
 ! Rainbow Roads (w Timid Soul)           (# 2) (3:10)
 ! Peach (w Diana Shroomy)                (# 3) (2:07)
 ! Ash (w Libano)                         (# 4) (2:29)
 ! Whispy Woods/Game Over (w Strider Kun) (# 5) (5:37)
 ! Been Thinking About It                 (# 6) (0:46)
 ! I Miss You (w ローマンRoman)               (# 7) (2:46)
 ! Dark City                              (# 8) (3:26)
 ! In The Moonlight                       (# 9) (2:13)
 ! ごめんなさい                                 (#10) (2:32)
 ! Lost Without You                       (#11) (1:22)
 ! Perfect Blue                           (#13) (2:55)
 ! [unknown]                              (#14)
```
See how track 12 is missing? It isn't missing on the website itself. Is there any way I can fix this?

**Comment by sampsyo** [comment, marked as answer]
"Missing tracks" means that you don't have those files.

**Reply by fetkitty** [reply]
I don't quite understand what you mean... I do have the files and they open in other programs.

**Reply by fetkitty** [reply]
I just tried redownloading one of the song files and the error still occurs. Only the song I want to select is missing from musicbrainz/spotify.

**Reply by sampsyo** [reply]
Here, beets is saying that this directory only contains one audio file:

```
C:\Users\User\Music\flac\Cham! (1 items)
```

Maybe try listing that directory?

**Reply by fetkitty** [reply]
Alright I tried doing this and the same thing happened. Here's the output:

```
beet import "C:\Users\User\Music\flac\Cham!\MACROSS 82-99 - Fun Tonight.flac"

C:\Users\User\Music\flac\Cham!\MACROSS 82-99 - Fun Tonight.flac (1 items)
Finding tags for album "マクロスMACROSS 82-99 - CHAM!".
Candidates:
1. マクロスMACROSS 82–99 - CHAM! (59.2%) (missing tracks) (Digital Media, 2017, XW, Neoncity Records)
2. マクロスMACROSS 82–99 - Aogashima Island ! (43.6%) (id, album, tracks) (Digital Media, 2017, XW)
3. マクロスMACROSS 82–99 - Sakura (41.0%) (id, album, tracks, ...) (Digital Media, 2017, XW, Banana Network)
4. マクロスMACROSS 82–99 - CHAM! (39.6%) (missing tracks, id, year) (Digital Media, 2015, XW)
5. マクロスMACROSS 82–99 - CHAM! (36.3%) (missing tracks, id, label, ...) (Digital Media, 2015, XW, DaFuture, DFTR008)
```
Any ideas?

**Reply by fetkitty** [reply]
Also the directory only contains one file on purpose, it's just one song not the whole album

**Reply by sampsyo** [reply]
If the directory only contains one song, then beets will say "you are missing the rest of the tracks on this album." That's what "missing tracks" means.

If you only want to import one song, not the whole album, then you might want a singleton (`-s`) import. The autotagger guide has more details:
https://beets.readthedocs.io/en/stable/guides/tagger.html

**Reply by fetkitty** [reply]
Ah great thank you now it works! I have one more question that I would like to ask:

Is there any way to use Spotify first and Musicbrainz second? I don't want to open a whole new discussion just for one question so i'll just ask here.

Thank you again for helping me out!

**Reply by arsaboo** [reply]
@fetkitty You can use `source_weight` to configure the desired order.
